From 503b0a90d1af91b0de86a323fe5ad9b33ebc4271 Mon Sep 17 00:00:00 2001
From: sgnanase <sundar.gnanasekaran@intel.com>
Date: Tue, 5 Oct 2021 21:20:28 +0530
Subject: [PATCH] Pointing flex and bison to prebuilts

This patch points the m4, flex and bison to the prebuilts
binary.

Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
---
 Android.mk                       | 3 ++-
 src/compiler/Android.glsl.gen.mk | 4 ++--
 src/freedreno/Android.ir3.mk     | 4 ++--
 src/mesa/program/Android.mk      | 2 +-
 4 files changed, 7 insertions(+), 6 deletions(-)

diff --git a/Android.mk b/Android.mk
index c63b6d8af49..6cf40263b1a 100644
--- a/Android.mk
+++ b/Android.mk
@@ -45,7 +45,8 @@ MESA_COMMON_MK := $(MESA_TOP)/Android.common.mk
 MESA_PYTHON2 := PYTHONPATH=$(PWD)/development/python-packages:$$PYTHONPATH python
 MESA_PYTHON3 := PYTHONPATH=$(PWD)/development/python-packages:$$PYTHONPATH python3
 ifeq ($(filter 5 6 7 8 9 10, $(MESA_ANDROID_MAJOR_VERSION)),)
-MESA_LEX     := M4=$(M4) $(LEX)
+MESA_LEX     := M4=$(PWD)/prebuilts/build-tools/linux-x86/bin/m4 $(PWD)/prebuilts/build-tools/linux-x86/bin/flex
+MESA_BISON     := M4=$(PWD)/prebuilts/build-tools/linux-x86/bin/m4 $(PWD)/prebuilts/build-tools/linux-x86/bin/bison
 else
 MESA_LEX     := $(LEX)
 endif
diff --git a/src/compiler/Android.glsl.gen.mk b/src/compiler/Android.glsl.gen.mk
index f654c869f83..ca19399b733 100644
--- a/src/compiler/Android.glsl.gen.mk
+++ b/src/compiler/Android.glsl.gen.mk
@@ -59,7 +59,7 @@ endef
 define glsl_local-y-to-c-and-h
 	@mkdir -p $(dir $@)
 	@echo "Mesa Yacc: $(PRIVATE_MODULE) <= $<"
-	$(hide) $(YACC) -o $@ -p "glcpp_parser_" $<
+	$(hide) $(MESA_BISON) -d -o $@ -p "glcpp_parser_" $<
 endef
 
 YACC_HEADER_SUFFIX := .hpp
@@ -67,7 +67,7 @@ YACC_HEADER_SUFFIX := .hpp
 define local-yy-to-cpp-and-h
 	@mkdir -p $(dir $@)
 	@echo "Mesa Yacc: $(PRIVATE_MODULE) <= $<"
-	$(hide) $(YACC) -p "_mesa_glsl_" -o $@ $<
+	$(hide) $(MESA_BISON) -d -p "_mesa_glsl_" -o $@ $<
 	touch $(@:$1=$(YACC_HEADER_SUFFIX))
 	echo '#ifndef '$(@F:$1=_h) > $(@:$1=.h)
 	echo '#define '$(@F:$1=_h) >> $(@:$1=.h)
diff --git a/src/freedreno/Android.ir3.mk b/src/freedreno/Android.ir3.mk
index 0fbb8c50ca6..176547e5c27 100644
--- a/src/freedreno/Android.ir3.mk
+++ b/src/freedreno/Android.ir3.mk
@@ -91,12 +91,12 @@ $(intermediates)/ir3/ir3_nir_trig.c: $(ir3_nir_trig_deps)
 $(intermediates)/ir3/ir3_parser.c: $(ir3_parser_deps)
 	@mkdir -p $(dir $@)
 	@echo "Gen Header: $(PRIVATE_MODULE) <= $(notdir $(@))"
-	$(hide) $(BISON) $< --name-prefix=ir3_yy --output=$@
+	$(hide) $(MESA_BISON) $< --name-prefix=ir3_yy --output=$@
 
 $(intermediates)/ir3/ir3_parser.h: $(ir3_parser_deps)
 	@mkdir -p $(dir $@)
 	@echo "Gen Header: $(PRIVATE_MODULE) <= $(notdir $(@))"
-	$(hide) $(BISON) $< --name-prefix=ir3_yy --defines=$@
+	$(hide) $(MESA_BISON) $< --name-prefix=ir3_yy --defines=$@
 
 include $(MESA_COMMON_MK)
 include $(BUILD_STATIC_LIBRARY)
diff --git a/src/mesa/program/Android.mk b/src/mesa/program/Android.mk
index 6b4e1916747..345fa30aa2a 100644
--- a/src/mesa/program/Android.mk
+++ b/src/mesa/program/Android.mk
@@ -29,7 +29,7 @@ endef
 define mesa_local-y-to-c-and-h
 	@mkdir -p $(dir $@)
 	@echo "Mesa Yacc: $(PRIVATE_MODULE) <= $<"
-	$(hide) $(YACC) -o $@ -p "_mesa_program_" $<
+	$(hide) $(MESA_BISON) -d -o $@ -p "_mesa_program_" $<
 endef
 
 # ----------------------------------------------------------------------
-- 
2.33.0

